### PR TITLE
fix(source-tracking): clean up mpd to prevent source tracking lib to gather status of all files

### PR DESCRIPTION
**Pull request by azlam-abdulsalam**
#### Description

Possibly source tracking lib use statusMatrix to gain information about all files when the
repository is initiated. This is gathered from the projects path at the point when the library is
intiatlized. As sfpowerscripts fixes source tracking one after the another, this is not necessary
and is resulting in some errors (statusMatrix unable to find files.. though no functional impact)
and also unnecessary computation. This fix is to clean up the sfdx-project.json to only carry the
package that is currently being fixed up for source tracking

#### Checklist
All items have to be completed before a PR is merged

- [x] Adhere to [Contribution Guidelines](https://docs.dxatscale.io/about-us/contributing-to-dx-scale)
- [x] Updates to Decision Records considered?
- [x] Updates to documentation at [DX@Scale Guide](https://github.com/dxatscale/dxatscale-guide) considered?
- [x] Tested changes?
- [x] Unit Tests new and existing passing locally?

### CODE REVIEW

File: packages/core/src/scratchorg/pool/ClientSourceTracking.ts [hunk 1]
```diff
@@ -60,31 +60,49 @@ export default class ClientSourceTracking {
...
+                SFPLogger.log(EOL, LoggerLevel.INFO, this.logger);
+                SFPLogger.log(
+                    COLOR_HEADER(`Package ${count} of ${sfpowerscriptsArtifacts.length}`),
+                    LoggerLevel.INFO,
+                    this.logger
+                );
+                SFPLogger.log(`Analyzing package ${COLOR_KEY_MESSAGE(artifact.Name)}`, LoggerLevel.INFO, this.logger);
+
+                //Reset the project json before going for full track
+                await git.raw(['restore', 'sfdx-project.json']);
```
Comment (aly76) [inline packages/core/src/scratchorg/pool/ClientSourceTracking.ts]: Cleaner if we just force checkout

File: packages/core/src/scratchorg/pool/ClientSourceTracking.ts [hunk 2]
```diff
@@ -60,31 +60,49 @@ export default class ClientSourceTracking {
...
+            this.cleanupSFDXProjectJsonTonOnePackage(tempDir.name, sfpowerscriptsArtifacts[0].Name);
+
             const project = await SfdxProject.resolve(tempDir.name);
 
             // Create local source tracking files in temp repo
             const tracking = await SourceTracking.create({
                 org: this.org,
                 project: project,
             });
-            tracking.ensureLocalTracking();
```
Comment (aly76) [inline packages/core/src/scratchorg/pool/ClientSourceTracking.ts:71]: Why is the initialization for source tracking removed?

Comment (azlam-abdulsalam) [reply packages/core/src/scratchorg/pool/ClientSourceTracking.ts:71]: Not required, as the updateLocalTracking calls it anyway

File: packages/core/src/scratchorg/pool/ClientSourceTracking.ts (hunk 1 above)
Comment (azlam-abdulsalam) [reply packages/core/src/scratchorg/pool/ClientSourceTracking.ts]: I looked at the faster option ( restore vs checkout ), as we only change one file

Comment (aly76) [reply packages/core/src/scratchorg/pool/ClientSourceTracking.ts]: It's probably faster to do a force checkout than two operations: restore and checkout

Comment (azlam-abdulsalam) [reply packages/core/src/scratchorg/pool/ClientSourceTracking.ts]: makes sense, fixing it

**Pull request merged**